pwiz_tools/Skyline/TestUtil/ScreenshotManager.cs

```diff
@@ -0,0 +1,309 @@
+        }
+
+        public class PointAdditive
+        {
+            private Point _add;
+            public PointAdditive(Point pAdd) { _add = pAdd; }
+            public PointAdditive(int pX, int pY) { _add = new Point(pX, pY); }
+            public static Point operator +(Point pt, PointAdditive pAdd) => new Point(pt.X + pAdd._add.X, pt.Y + pAdd._add.Y);
+            public static Size operator +(Size sz, PointAdditive pAdd) => new Size(sz.Width + pAdd._add.X, sz.Height + pAdd._add.Y);
+        }
+        public enum ShotType{ ActiveWindow, SkylineWindow, SkylineCustomArea}
+
+        private abstract class SkylineScreenshot
+        {
+            private ShotType _type;
+            private SkylineWindow _skylineWindow;
+
+            protected const string SHOT_ELEMENT = "shot";
+            protected const string SHOT_TYPE_ATTRIBUTE = "type";
+            protected const string SHOT_TYPE_VAL_ACTIVE_FORM = "active_form";
+            protected const string SHOT_TYPE_VAL_CUSTOM_AREA = "skyline_relative_frame";
+            protected const string SHOT_FRAME_ELEMENT = "frame";
+            protected const string SHOT_FRAME_LEFT_ATTRIBUTE = "left";
+            protected const string SHOT_FRAME_TOP_ATTRIBUTE = "top";
+            protected const string SHOT_FRAME_RIGHT_ATTRIBUTE = "right";
+            protected const string SHOT_FRAME_BOTTOM_ATTRIBUTE = "bottom";
+
+
+            [DllImport("gdi32.dll")]
+            static extern int GetDeviceCaps(IntPtr hdc, int nIndex);
+            [DllImport("user32.dll")]
+            private static extern IntPtr GetForegroundWindow();
+
+            public enum DeviceCap
+            {
+                VERTRES = 10,
+                DESKTOPVERTRES = 117,
+            }
+            
+            /**
+             * Factory method
+             */
+            public static SkylineScreenshot createScreenshot(SkylineWindow pSkylineWindow, [NotNull] XmlNode shotNode)
+            {
+                // ReSharper disable PossibleNullReferenceException
+                if (shotNode.Attributes[SHOT_TYPE_ATTRIBUTE] == null)
+                    throw new InvalidDataException("Invalid XML. type attribute was expected but was not found.");
+
+                if (shotNode.Attributes[SHOT_TYPE_ATTRIBUTE].Value == SHOT_TYPE_VAL_ACTIVE_FORM)
+                    return new ActiveWindowShot(pSkylineWindow, shotNode);
+                else if (shotNode.Attributes[SHOT_TYPE_ATTRIBUTE].Value == SHOT_TYPE_VAL_CUSTOM_AREA)
+                    return new CustomAreaShot(pSkylineWindow, shotNode);
+                else  throw new InvalidDataException("Unsupported screenshot type");
+                // ReSharper enable PossibleNullReferenceException
+            }
+
+            public SkylineScreenshot(ShotType pShotType, SkylineWindow pSkylineWindow)
+            {
+                _type = pShotType;
+                _skylineWindow = pSkylineWindow;
+            }
+
+            protected float getScalingFactor()
```

Comment:
This function name should be capitalized.
Also, it seems like maybe this method should return a "PointFactor" instead of a float

Reply:
Agree.